homeassistant/components/homematicip_cloud/sensor.py

```diff
@@ -102,116 +245,102 @@ async def async_setup_entry(
+
+    # Handle floor terminal blocks separately
+    floor_terminal_blocks = (
+        FloorTerminalBlock6,
+        FloorTerminalBlock10,
+        FloorTerminalBlock12,
+        WiredFloorTerminalBlock12,
+    )
+    entities.extend(
+        HomematicipFloorTerminalBlockMechanicChannelValve(
+            hap, device, channel=channel.index
+        )
+        for device in hap.home.devices
+        if isinstance(device, floor_terminal_blocks)
+        for channel in device.functionalChannels
+        if isinstance(channel, FloorTerminalBlockMechanicChannel)
+        and getattr(channel, "valvePosition", None) is not None
+    )
 
     async_add_entities(entities)
 
 
+class HomematicipTiltAngleSensor(HomematicipGenericEntity, SensorEntity):
+    """Representation of the HomematicIP tilt angle sensor."""
+
+    _attr_native_unit_of_measurement = DEGREE
+    _attr_state_class = SensorStateClass.MEASUREMENT_ANGLE
+
+    def __init__(self, hap: HomematicipHAP, device) -> None:
+        """Initialize the tilt angle sensor device."""
+        super().__init__(hap, device, post="Tilt Angle")
+
+    @property
+    def native_value(self) -> int | None:
+        """Return the state."""
+
+        if hasattr(self.functional_channel, "absoluteAngle"):
+            return (
+                self.functional_channel.absoluteAngle
+                if self.functional_channel.absoluteAngle is not None
+                else None
+            )
+
+        return None
+
+
+class HomematicipTiltStateSensor(HomematicipGenericEntity, SensorEntity):
+    """Representation of the HomematicIP tilt sensor."""
+
+    _attr_device_class = SensorDeviceClass.ENUM
+    _attr_options = TILT_STATE_VALUES
+    _attr_translation_key = "tilt_state"
+
+    def __init__(self, hap: HomematicipHAP, device) -> None:
+        """Initialize the tilt sensor device."""
+        super().__init__(hap, device, post="Tilt State")
+
+    @property
+    def native_value(self) -> str | None:
+        """Return the state."""
+        if hasattr(self.functional_channel, "tiltState"):
+            return (
+                self.functional_channel.tiltState.lower()
+                if self.functional_channel.tiltState is not None
+                else None
+            )
+
+        return None
+
+    @property
+    def extra_state_attributes(self) -> dict[str, Any]:
+        """Return the state attributes of the tilt sensor."""
+        state_attr = super().extra_state_attributes
+
+        state_attr[ATTR_ACCELERATION_SENSOR_NEUTRAL_POSITION] = getattr(
+            self.functional_channel, "accelerationSensorNeutralPosition", None
+        )
+        state_attr[ATTR_ACCELERATION_SENSOR_TRIGGER_ANGLE] = getattr(
+            self.functional_channel, "accelerationSensorTriggerAngle", None
+        )
+        state_attr[ATTR_ACCELERATION_SENSOR_SECOND_TRIGGER_ANGLE] = getattr(
+            self.functional_channel, "accelerationSensorSecondTriggerAngle", None
+        )
+
+        return state_attr
```

Comment:
can these be changed? Can we make separate entities out of them?

Reply:
these are just config values. Make separate entities anyway?